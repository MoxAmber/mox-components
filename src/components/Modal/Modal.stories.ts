import type { Meta, StoryObj } from "@storybook/web-components";
import { html } from "lit";
import ".";

const meta: Meta = {
  component: "mox-modal",
};
export default meta;

export const Primary: StoryObj = {
  render: () => html`
    <button
      type="button"
      @click=${() => document.querySelector("mox-modal")?.showModal()}
    >
      Open Modal
    </button>
    <mox-modal>
      <h1 slot="title">Modal title</h1>
      <p slot="body">And this is some content</p>
    </mox-modal>
  `,
};
