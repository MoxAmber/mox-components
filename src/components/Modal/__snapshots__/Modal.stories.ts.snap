// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Modal ClickOutside play-test: shadow DOM 1`] = `
<dialog part="dialog">
  <header>
    <slot name="title">
    </slot>
    <button type="button"
            part="close"
            aria-label="Close"
            autofocus
    >
      X
    </button>
  </header>
  <slot>
  </slot>
</dialog>
`;

exports[`components/Modal Main play-test: shadow DOM 1`] = `
<dialog part="dialog"
        open
>
  <header>
    <slot name="title">
    </slot>
    <button type="button"
            part="close"
            aria-label="Close"
            autofocus
    >
      X
    </button>
  </header>
  <slot>
  </slot>
</dialog>
`;
